solutions/cf-1208F.mdx

```diff
@@ -0,0 +1,74 @@
+---
+id: cf-1208F
+source: CF
+title: Bits and Pieces
+author: Dong Liu
+---
+
+## Explanation
+
+Let $|$ denote the bitwise OR operation and $\&$ denote the bitwise AND operation.
+
+The answer is equivalent of enumerating $a_i$ and finding the optimal $x$ where $x=a_j\&a_k$ and $i<j<k$ and $a_i|x$ is maximum.
+
+Note that $x$ can be binary searched since $2^b>2^0+2^1+...+2^{b-1}$, so we can loop through each $2^b$ in decreasing order and check if there exists an answer for $z$ such that $z$ is a superset of $x|2^b$. When we reach a bit $2^b$ such that it is contained in $a_i$, it doesn't matter whether it is in the final $x$ since $a_i|x$ already contains $2^b$.
```

Comment:
```suggestion
Note that $x$ can be binary searched since $2^b>2^0+2^1+\dots+2^{b-1}$, so we can loop through each $2^b$ in decreasing order and check if there exists an answer for $z$ such that $z$ is a superset of $x|2^b$. When we reach a bit $2^b$ such that it is contained in $a_i$, it doesn't matter whether it is in the final $x$ since $a_i|x$ already contains $2^b$.
```